app/experimenter/reporting/tasks.py

```diff
@@ -0,0 +1,216 @@
+        create_reportlog(n_changelog)
+
+    for experiment in live_experiments:
+        if experiment.normandy_id:
+            create_reportlog_from_normandy_history(experiment)
+
+
+def get_latest_log_date():
+    if ReportLog.objects.count():
+        return ReportLog.objects.latest("timestamp").timestamp
+    return timezone.datetime.min
+
+
+def create_reportlog(changelog):
+
+    experiment = changelog.experiment
+    experiment_type = get_experiment_type(experiment)
+    timestamp = changelog.changed_on
+    event = get_event_type(changelog)
+    event_reason = get_event_reason(changelog, event)
+    message = changelog.message or ""
+
+    if event and not is_duplicate_recipe_change(timestamp, experiment, event_reason):
+
+        report_log = ReportLog.objects.create(
+            timestamp=timestamp,
+            experiment_slug=experiment.slug,
+            experiment_name=experiment.name,
+            experiment_type=experiment_type,
+            experiment_old_status=changelog.old_status,
+            experiment_new_status=changelog.new_status,
+            event=event,
+            event_reason=event_reason,
+            comment=message,
+        )
+        report_log.projects.set(experiment.projects.all())
+
+
+def create_reportlog_from_normandy_history(experiment):
+
+    history = normandy.get_history_list(experiment.normandy_id)
+
+    # first normandy history revision is creation
+    for revision in history[:-1]:
+        timestamp = parser.parse(revision.get("updated"))
+        event_reason = ReportLog.EventReason.RECIPE_CHANGE
+        experiment_type = get_experiment_type(experiment)
+        event = ReportLog.Event.UPDATE
+        message = ""
+        if ReportLog.objects.filter(
+            timestamp__date=timestamp,
+            experiment_name=experiment.name,
+            event_reason=event_reason,
+        ).exists():
+            break
+
+        report_log = ReportLog.objects.create(
+            timestamp=timestamp,
+            experiment_slug=experiment.slug,
+            experiment_name=experiment.name,
+            experiment_type=experiment_type,
+            experiment_old_status=ReportLog.ExperimentStatus.LIVE,
+            experiment_new_status=ReportLog.ExperimentStatus.LIVE,
+            event=event,
+            event_reason=event_reason,
+            comment=message,
+        )
+        report_log.projects.set(experiment.projects.all())
+
+
+def is_duplicate_recipe_change(date, experiment, event_reason):
+    if event_reason == ReportLog.EventReason.RECIPE_CHANGE:
+        return ReportLog.objects.filter(
+            timestamp__date=date,
+            experiment_name=experiment.name,
+            event_reason=ReportLog.EventReason.RECIPE_CHANGE,
+        ).exists()
+    return False
+
+
+def get_experiment_type(experiment):
+    if type(experiment) == NimbusExperiment:
+        application = experiment.application
+        application_slug = NimbusConstants.APPLICATION_CONFIGS[application].slug
```

Comment:
There's already a method for this:

https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/models/nimbus.py#L225-L228